File: src/php/Generic/StringManipulationBase.php
<?php

namespace EntityFX\NetBenchmark\Core\Generic {

    use EntityFX\NetBenchmark\Core\Writer;
    
    abstract class StringManipulationBase  extends BenchmarkBase {
        public function __construct($writer, $printToConsole)
        {
            parent::__construct($writer, $printToConsole);
            $this->Iterrations = 5000000;
            $this->Ratio = 25;
        }

        public static function DoStringManipilation($str)
        {
            return str_replace("aaa", ".", strtolower(
                    strtoupper(
                        str_replace("/", "_", implode("/", explode(' ', $str)))
                    ) . "AAA"
                )
            );
        }
    }
}

File: src/lua/generic/stringManipulationBase.lua
StringManipulationBase = class(BenchmarkBase, function(a, writer, printToConsole)
    BenchmarkBase.init(a, writer, printToConsole)
    a.iterrations = 5000000
    a.ratio = 10
end)

function StringManipulationBase:doStringManipilation(str)
    return string.lower(string.upper(table.concat(split(str, ' '), "/"):gsub('/', ' ')) .. "AAA"):gsub('aaa', '.')
end